src/util-inl.h

```diff
@@ -229,30 +229,74 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+template <typename T>
+T MultiplyWithOverflowCheck(T a, T b) {
```

Comment:
I don't think making this a template is a good idea. It can't be used with signed types since signed overflow is UB, so the check could be optimized out.